### ISSUE TITLE
decrypt ask always for discovery switch

### ISSUE THREAD

**Issue by antoweb** [aws/aws-encryption-sdk-cli#186]
Version 2.0.0 encrypt works good but whe i decrypt folowing error message appear:

aws-encryption-cli: error: Discovery must be set to True or False.

Even specifieng or not wrapping-keys, even using or not discovery=true or fals

**Comment by maxtess** [comment]
+1

**Comment by antonf-amzn** [comment]
Hi,

What provider are you using?  I have verified that it works for `aws-kms` provider.  

Also, you have to explicitly add `discovery=False` for it to work, otherwise you will see the `Discovery must be set to True or False.`

For example:

`aws-encryption-cli -w provider=aws-kms key=arn:some_kms_arn -S -d -i data.encrypted -o data.decrypted --discovery=true`

**Comment by antoweb** [comment]
Hello,
i'm not specifing provider my command is like the command documented in aws site: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/crypto-cli-how-to.html on the chapter "How to encrypt and decrypt data" i launch the following commands (like aws example)

aws-encryption-cli --encrypt --input pippo.txt --wrapping-keys key=MY-KEY region=eu-west-1 --output pippo.encrypted --metadata-output ~/metadata --encryption-context purpose=test --commitment-policy require-encrypt-require-decrypt

aws-encryption-cli --decrypt --input pippo.encrypted --wrapping-keys key=MY-KEY region=eu-west-1 --output pippo.plain --metadata-output ~/metadata --encryption-context purpose=test --commitment-policy require-encrypt-require-decrypt

on the second command (decrypt) the error received is:

aws-encryption-cli: error: Discovery must be set to True or False.

**Comment by antonf-amzn** [comment]
Hi,

I think I understand the problem.

There is a problem with our documentation that is unclear about the `--discovery` flag.  We will work on fixing this and providing clearer documentation.

To fix your immediate problem, please add `--discovery=False` to your command.

**Comment by farleyb-amazon** [comment]
Hi antoweb,

We have released new versions of the AWS Encryption CLI and updated the [documentation](https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/crypto-cli-how-to.html) that explains how to enable discovery mode. There was a bug in previous versions of the CLI which did not correctly implement strict and discovery modes.

In version 1.8 and 2.1, we have replaced the `--discovery` parameter with the `discovery` attribute of the `--wrapping-keys` parameter. In version 2.1 the `discovery` attribute is only required if you are not specifying wrapping keys with the `key` attribute, so the original examples you gave should now work with version 2.1 of the CLI:

`
aws-encryption-cli --decrypt --input pippo.encrypted --wrapping-keys key=MY-KEY region=eu-west-1 --output pippo.plain --metadata-output ~/metadata --encryption-context purpose=test --commitment-policy require-encrypt-require-decrypt
`

Note that MY-KEY in this case needs to be the key ARN, not a key alias.
